### DISCUSSION TITLE
Should `/etc/default/grub` be recognized as INI?

### DISCUSSION THREAD

**Discussion by Rudxain** [helix-editor/helix#11217] [Ideas]
When I was viewing that file in `hx`, I had to `:set-language ini` for it to recognize the format. The title is a question because I'm unsure if it's actually INI. I'm also asking because it may be a good idea to add a `{ glob = "grub" }` to the default `languages.toml`

**Comment by TornaxO7** [comment]
According to the [official grub config documentation](https://www.gnu.org/software/grub/manual/grub/html_node/Simple-configuration.html) (I just skimmed through it) it says that the `/etc/default/grub` file only contains key-value entries:

> The file /etc/default/grub controls the operation of grub-mkconfig. It is sourced by a shell script, and so must be valid POSIX shell input; normally, it will just be a sequence of ‘KEY=value’ lines, but if the value contains spaces or other special characters then it must be quoted.

However, files in `/etc/grub.d/` seem to have their own syntax.

**Reply by Rudxain** [reply]
That explains a lot! So it's like a "shell profile" rather than a script?

**Reply by TornaxO7** [reply]
May I ask what a shell profile is? :eyes:

**Reply by Rudxain** [reply]
> May I ask what a shell profile is? 👀

`*rc` files! such as:
- `.bashrc`
- `.mkshrc`
- `.profile` (POSIX)
- etc...

I'm asking this because my `grub` file is a valid POSIX-shell script (according to `shellcheck`). It even has a line that looks like *command-substitution*:
```sh
GRUB_DISTRIBUTOR=`lsb_release -i -s 2> /dev/null || echo Debian`
```

**Reply by TornaxO7** [reply]
Oh damn...

didn't knew that you could write shell commands inside `grub`.

![peepoHeavyNotes](https://github.com/user-attachments/assets/1fbfe699-21f3-4615-9ec4-f6f9f3f196e0)

I think it's better to wait for someone who knows more than me.
